src/main/java/net/minecraftforge/registries/GameData.java

```diff
@@ -964,4 +996,26 @@ private static void forceRegistryName(IForgeRegistryEntry<?> entry, ResourceLoca
         }
 
     }
+
+    private static class AttributeSetupState
```

Comment:
Currently the way the loading stages work it does not seem possible to hold the state on the event itself without functionally doing the equivalent of what I am doing here. The suppliers are re-invoked for each stage. This works ok in the one place where it's really used because the registries hold the state instead. To be honest I was somewhat surprised that the post stage did not receive the same stream of events as passed into the stage itself. 

I would really appreciate some more guidance on what ya'll want that to look like because the functionality does not exist right now. The only reason I am doing the "thing that is bad" is because there is no other way to achieve that thing right now and it seemed even uglier to do some kind of supplier that stores the state inside itself (and would mostly be logically the exact same thing).